pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountCPCSketchAggregationFunction.java

```diff
@@ -422,16 +422,18 @@ public Comparable mergeFinalResult(Comparable finalResult1, Comparable finalResu
 
   @Override
   public boolean canUseStarTree(Map<String, Object> functionParameters) {
-    Object lgK = functionParameters.get(Constants.CPCSKETCH_LGK_KEY);
+    Object lgKParam = functionParameters.get(Constants.CPCSKETCH_LGK_KEY);
+    int starTreeLgK;
 
-    // Check if lgK values match
-    if (lgK != null) {
-      return _lgNominalEntries == Integer.parseInt(String.valueOf(lgK));
+    if (lgKParam != null) {
+      starTreeLgK = Integer.parseInt(String.valueOf(lgKParam));
     } else {
       // If the functionParameters don't have an explicit lgK set, it means that the star-tree index was built with
       // the default value for lgK
-      return _lgNominalEntries == CommonConstants.Helix.DEFAULT_CPC_SKETCH_LGK;
+      starTreeLgK = CommonConstants.Helix.DEFAULT_CPC_SKETCH_LGK;
     }
+    // Check if the query lgK param is less than or equal to that of the StarTree aggregation
```

Comment:
nit: a one-liner explanation on _why_ we're doing a <= check (https://github.com/apache/pinot/pull/13835#discussion_r1787334771) might be useful for future readers here.

Reply:
done.